sdk/trafficmanager/Azure.ResourceManager.TrafficManager/src/Customization/EndpointCollection.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.ResourceManager.TrafficManager
+{
+    /// <summary>
+    /// The class to overcome issue with the Endpoint Collection REST API where there is no REST API couterpart that GETs all
+    /// profile data resource. The all profile data resources are retrived from the collection of endpoints attached to <see cref="ProfileData"/>.
+    /// </summary>
+    public partial class EndpointCollection : ArmCollection
+    {
+        private readonly ProfileData _profileData;
+
+        /// <summary> Initializes a new instance of the <see cref="EndpointCollection"/> class. </summary>
+        /// <param name="client"> The client parameters to use in these operations. </param>
+        /// <param name="id"> The identifier of the parent resource that is the target of operations. </param>
+        /// /// <param name="profileData">The parent profile data. </param>
+        internal EndpointCollection(ArmClient client, ResourceIdentifier id, ProfileData profileData) : this(client, id)
+        {
+            Argument.AssertNull(profileData, nameof(profileData));
+
+            this._profileData = profileData;
+        }
+
+        /// <summary>
+        /// Asynchroniosly lists all Traffic Manager endpoints within a profile.
+        /// </summary>
+        /// <returns> A collection of <see cref="EndpointResource" /> that may take multiple service requests to iterate over. </returns>
+        public virtual AsyncPageable<EndpointData> GetAllAsync()
```

Comment:
this is marked in the md as list exception yet we are adding it here?  Without digging in more I think we should remove that exception and simply leave this custom implementation if that is the difference that way the collection will have the property interfaces added like `IEnumerable`

Reply:
Unfortunately, the exception and the customization were added on purpose and without the exception code generation does not succeed.

This is what node returns when we remove the exception from md.

OpenAPI spec has a security definition but autorest security config is defined. Security config from autorest will be used.
fatal   | The ResourceCollection EndpointCollection (RequestPath: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/trafficmanagerprofiles/{profileName}/{endpointType}/{endpointName}) does not have a GetAll method

Reply:
Lets add the interfaces to the customized class in that case